features/include/pcl/features/good.h

```diff
@@ -0,0 +1,385 @@
+      {
+	center_of_bbox = center_of_bbox_;
+      }
+
+      /** \brief get dimensions of bounding box of a set of points given by setInputCloud()
+        * \param[out] bbox_dimensions the resultant boundingbox dimensions
+        */ 
+      inline void
+      getObjectBoundingBoxDimensions(pcl::PointXYZ &bbox_dimensions) const
+      {
+	bbox_dimensions = bbox_dimensions_;
+      }
+
+      /** \brief get the order of protection plans in constructing GOOD descriptor
+        * \param[out] order_of_projected_plane the resultant of order of projections
+        */  
+      inline void
+      getOrderOfProjectedPlanes (std::string &order_of_projected_plane) const
+      {
+	order_of_projected_plane = order_of_projected_plane_;
+      }
+
+      /** \brief get the transformation matrix from camera reference frame to object local reference frame
+        * \param[out] transformation the resultant transformation matrix
+        */       
+      inline void
+      getTransformationMatrix (Eigen::Matrix4f &transformation) const
+      {
+	transformation = transformation_;
+      } 
+
+    
+    protected:
+
+      /** \brief Estimate the GOOD descriptor at a set of points given by setInputCloud() 
+      * \param[out] object_description  the resultant GOOD descriptor representing the feature at the query point cloud
+      */        
+
+      virtual void
+      computeFeature (PointCloudOut &output);
+    
+
+    private:
+
+      /** \brief number of bins along one dimension; each projection plane is divided into number_of_bins ? number_of_bins square bins. 
+       * By default, the number_of_bins_ is set to 5.
+       */
+      unsigned int number_of_bins_; 
+      
+      /** \brief threshold parameter is used in constructing local reference frame. 
+       * By default, the number_of_bins_ is set to 0.0015.
+       */
+      float threshold_;
+
+      /** \brief resultant of sign disambiguation can be either 1 or -1 */
+      int sign_;
+            
+      /** \brief transformed point cloud in LRF */
+      PointCloudIn transformed_point_cloud_;  
+      
+      /** \brief get transformation matrix */
+      Eigen::Matrix4f transformation_;
+      
+      /** \brief dimensions of boundingboxbox of given point cloud */
+      pcl::PointXYZ bbox_dimensions_;
+      
+      /** \brief center of boundingboxbox of given point cloud */
+      pcl::PointXYZ center_of_bbox_;
+      
+      /** \brief vector of three point clouds containing orthographic projection views */
+      std::vector < PointCloudIn > vector_of_projected_views_;
+      
+      /** \brief get order of projection views e.g. XoY-XoZ-YoZ */
+      std::string order_of_projected_plane_;    
+      
+      /** \brief get dimensions of bounding box of a given point cloud
+ 	* \param[in] pc pointer to a point cloud.
+	* \param[out] dimensions the resultant boundingbox dimensions
+        */ 
+      void 
+      computeBoundingBoxDimensions (PointCloudIn pc, pcl::PointXYZ &dimensions);

```

Comment:
In fact, there is no reason to pass the point cloud by pointer. Note that `pcl::getMinMax3D` that you use inside this function takes its argument by reference. This is a good style to not impose unnecessary requirements on storage type of your arguments.

This [post](https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/) is an excellent reading on the topic.